tests/reftests/rec-pin.test

```diff
@@ -0,0 +1,994 @@
+N0REP0
+### OPAMYES=1 OPAMWORKINGDIR=0
+### # structure
+### : $ find pinnes
+### : .pin1.opam
+### : ./a
+### : ./a/pin1sub1.opam
+### : ./a/i
+### : ./a/i/pin1sub1sub1.opam
+### : ./a/i/j
+### : ./a/i/j/pin1sub1sub1sub1.opam
+### : ./pin1.opam
+### : ./a/k
+### : ./a/k/pin1sub1sub2.opam
+### : ./b
+### : ./b/pin1sub2.opam
+### <pin:pinnes/pin1.opam>
+
+### <pin:pinnes/a/pin1sub1.opam>
+
+### <pin:pinnes/a/i/pin1sub1sub1.opam>
+
+### <pin:pinnes/a/i/j/pin1sub1sub1sub1.opam>
+
+### <pin:pinnes/a/k/pin1sub1sub2.opam>
+
+### <pin:pinnes/b/pin1sub2.opam>
+
+### git -C ./pinnes init -q
+### git -C ./pinnes config core.autocrlf false
+### git -C ./pinnes add a/i b
+### git -C ./pinnes commit -qm "fst"
+### opam switch create devnull --empty
+### opam pin -n --recursive pinnes
+This will pin the following packages: pin1, pin1sub1, pin1sub1sub1, pin1sub1sub1sub1, pin1sub1sub2, pin1sub2. Continue? [Y/n] y
```

Comment:
Something seems wrong here.
Above we only add `a/i` and `b` to the git directory. Git is supposed to be used by default, however when pinning the opam files that are not supposed to be in the git repository yet are also pinned (`pin1`, `pin1sub1` and `pin1sub1sub2`)

Reply:
Replying to myself after testing. This is actually the current behaviour of opam pin (scan the directory for opam file, even if it is not in the git repository).

Fair enough.

Reply:
yes there is the implicit behaviour that even if the opam file isn't committed, it is taken into account.